06_puppy/madhavansu/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+/**
+* Storer implementation in below scenarios
+* Dynamic Map ID: MapStore implementation of Storer backed by a map
+* Static Map ID: SyncStore implementation of Storer backed by a sync.Map
+ */
+//Sync store
+type syncStore struct {
+	sync.Mutex
+	sync.Map
+}
+
+func newSyncStore() *syncStore {
+	return &syncStore{}
+}
+
+func (s *syncStore) createPuppy(in Puppy) uint {
+	s.Lock()
+	defer s.Unlock()
+	s.Store(in.id, in)
+	return in.id
+}
+
+func (s *syncStore) readPuppy(id uint) Puppy {
+	pd, ok := s.Load(id)
+	if !ok {
+		return Puppy{}
+	}
+	p, _ := pd.(Puppy)
+	return p
+}
+
+func (s *syncStore) updatePuppy(id uint, in Puppy) {
+	s.Store(in.id, in)
+}
+
+func (s *syncStore) deletePuppy(id uint) {
+	s.Lock()
+	defer s.Unlock()
+	s.Delete(id)
```

Comment:
The `Delete` method for `sync.Map` is already thread-safe, you don't need to `Lock` and `Unlock` this call.

Reply:
Updated.